shared/colors.py

```diff
@@ -53,3 +51,31 @@ def get_switch_color(self, dpid):
             return self._colors[dpid]
         except KeyError:
             return {}
+
+    async def _aget_colors(self):
+        """Get list of colors asynchronously."""
+        async with httpx.AsyncClient() as client:
+            result = await client.get(self._url)
+            if result.is_server_error or result.status_code >= 400:
+                log.error(f'Error ocurred when getting colors: {result.text}')
```

Comment:
Should we return early after this error? 

`coloring` GET /colors is highly unlikely to fail, but always assuming that you'd be able to access `result['colors']` might eventually lead to issues down the road as the code evolves.

Reply:
Oh ye, added `return`